ts_src/psbt.ts

```diff
@@ -182,6 +182,17 @@ export class Psbt {
   }
 
   addInput(inputData: PsbtInputExtended): this {
+    if (
```

Comment:
Why is this check required?

Shouldn't TypeScript throw for us if `inputData` doesn't match the `PsbtInputExtended` interface?

Reply:
TypeScript is only a compile-time check (transpile-time actually).

Some users don't use TypeScript, so we should do runtime checks as well.

Reply:
Ah, I'm with you.

I wonder why TypeScript don't add this error checking automatically for non TypeScript consumers.

They could just represent all interfaces as JavaScript objects and then inject a checking function as the fist line of the body of all functions.

e.g:

```js
addInput(inputData) {
  checkTsInterface(inputData, PsbtInputExtended);
```

Reply:
That's what typeforce does... but the problem is that errors like that are extremely unhelpful.

If you look at the original issue, it amounts to "what does this error mean?" and it was basically the same, "Expected string, got Buffer" etc...

Saying a specifically tailored "hey, this method requires this type for this argument, and you didn't provide that." for each and every method is hard.